templater/jobs/presubmits.go

```diff
@@ -0,0 +1,275 @@
+package jobs
+
+type ExtraRef struct {
+	BaseRef string
+	Org     string
+	Repo    string
+}
+
+type EnvVar struct {
+	Name  string
+	Value string
+}
+
+type Resources struct {
+	Requests ResourceConfig
+	Limits   ResourceConfig
+}
+
+type ResourceConfig struct {
+	CPU    string
+	Memory string
+}
+
+type JobConfig struct {
+	JobName            string
+	RepoName           string
+	RunIfChanged       string
+	Timeout            string
+	ImageBuild         bool
+	ExtraRefs          []ExtraRef
+	ServiceAccountName string `default:"presubmits-build-account"`
+	EnvVars            []EnvVar
+	Commands           []string
+	Resources          Resources
+}
+
+func GetPresubmitList() map[string]JobConfig {
+	presubmitList := map[string]JobConfig{
+		"cni-presubmits.yaml": JobConfig{
+			JobName:      "cni-plugins-presubmit",
+			RepoName:     "aws/eks-distro",
+			RunIfChanged: "^build/lib/.*|Common.mk|projects/containernetworking/plugins/.*",
+			Commands:     []string{"make build -C projects/containernetworking/plugins"},
+			Resources: Resources{
+				Requests: ResourceConfig{
+					CPU:    "1024m",
+					Memory: "2Gi",
+				},
+				Limits: ResourceConfig{
+					CPU:    "1024m",
+					Memory: "2Gi",
+				},
+			},
+		},
+		"coredns-1-18-presubmits.yaml": JobConfig{
```

Comment:
TODO: Find a better way to generate these release-branch based jobs that only differ in the release branch component.